c/iotdm_client/lwm2m_objects/device_object.h

```diff
@@ -17,6 +17,13 @@
 
 #include "iothub_client.h"
 
+#ifdef __MACH__
```

Comment:
Cameron added these to get OSX build working.  Not sure why they aren't required in Linux/Windows build?  There are defined structs/functions in these headers which _are_ used in this file...we need to investigate why the build _doesn't_ fail in Linux/Windows.